### PR TITLE
Prepare release 3.1.2

### PR DESCRIPTION
**Pull request by autophagy**
## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: blackbox/docs/appendices/release-notes/3.1.2.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+.. NOTE::
+
+    If you are upgrading a cluster, you must be running CrateDB 2.0.4 or higher
+    before you upgrade to 3.1.2.
+
+    We recommend that you upgrade to the latest 3.0 release before moving to
+    3.1.2.
+
+    You cannot perform a `rolling upgrade`_ to this version. Any upgrade to this
```
Comment (mfussenegger) [inline blackbox/docs/appendices/release-notes/3.1.2.rst]: A rolling upgrade from `3.1.1` to `3.1.2` is supported

Comment (mfussenegger) [reply blackbox/docs/appendices/release-notes/3.1.2.rst]: 3.1.0 was a testing release and never marked as stable. That's why the information there is correct as we don't support rolling upgrade from testing release to stable release.
3.1.1 is stable, so a rolling upgrade from it should be possible.

File: blackbox/docs/appendices/release-notes/3.1.2.rst [hunk 2]
```diff
@@ -14,11 +14,9 @@ Released on 2018/10/18.
     We recommend that you upgrade to the latest 3.0 release before moving to
     3.1.2.
 
-    You cannot perform a `rolling upgrade`_ to this version. Any upgrade to this
-    version will require a `full restart upgrade`_.
-
-    When restarting, CrateDB will migrate indexes to a newer format. Depending
-    on the amount of data, this may delay node start-up time.
+    If you want to perform a `rolling upgrade`_, your current CrateDB version
+    number must be at least :ref:`version_3.1.0`. Any upgrade from a version
```
Comment (mfussenegger) [inline blackbox/docs/appendices/release-notes/3.1.2.rst]: `3.1.1` - (see my second response) - sorry, should've  clarified that in my first comment.

File: blackbox/docs/appendices/release-notes/3.1.2.rst [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+.. NOTE::
+
+    If you are upgrading a cluster, you must be running CrateDB 2.0.4 or higher
+    before you upgrade to 3.1.2.
+
+    We recommend that you upgrade to the latest 3.0 release before moving to
+    3.1.2.
+
+    If you want to perform a `rolling upgrade`_, your current CrateDB version
+    number must be at least :ref:`version_3.1.0`. Any upgrade from a version
```
Comment (mfussenegger) [inline blackbox/docs/appendices/release-notes/3.1.2.rst]:
```suggestion
    number must be at least :ref:`version_3.1.1`. Any upgrade from a version
```

File: blackbox/docs/appendices/release-notes/3.1.2.rst (hunk 1 above)
Comment (autophagy) [reply blackbox/docs/appendices/release-notes/3.1.2.rst]: Ah, that makes sense. Thank you. :slightly_smiling_face:

**Pull request merged**